app/src/main/java/com/example/abha_create_verify_android/AadhaarQRScanActivity.kt

```diff
@@ -0,0 +1,96 @@
+package com.example.abha_create_verify_android
+
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.example.abha_create_verify_android.PatientSubject.Companion.patientSubject
+import com.example.abha_create_verify_android.utils.AadhaarPlainTextQrParser
+import com.example.abha_create_verify_android.utils.AadhaarSecureQrParser
+import com.example.abha_create_verify_android.utils.AadhaarXmlQrParser
+import com.example.abha_create_verify_android.utils.Patient
+import com.example.abha_create_verify_android.utils.Variables
+import com.journeyapps.barcodescanner.ScanContract
+import com.journeyapps.barcodescanner.ScanIntentResult
+import com.journeyapps.barcodescanner.ScanOptions
+import org.xml.sax.InputSource
+import javax.xml.parsers.DocumentBuilderFactory
+
+
+class AadhaarQRScanActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        Variables.isCreateAbhaScan = intent.getBooleanExtra("isCreateAbhaScan", false)
+
+        barcodeLauncher.launch(ScanOptions().setOrientationLocked(false).setPrompt("Scan Aadhaar QR Code"))
+    }
+
+    private val barcodeLauncher = registerForActivityResult<ScanOptions, ScanIntentResult>(
+        ScanContract()
+    ) { result: ScanIntentResult ->
+        if (result.contents == null) {
+            Toast.makeText(this, "Cancelled", Toast.LENGTH_SHORT).show()
+            handleBack()
+        } else {
+            Toast.makeText(this, "Scanned, Processing... ", Toast.LENGTH_SHORT)
+                .show()
+            result.contents?.let {
+                processQRData(it)
+                displayAadhaarInfo()
+            }
+        }
+    }
+
+
+    private fun handleBack() {
+        if(Variables.isCreateAbhaScan) {
+            val intent = Intent(this, DemographicsManualOrQRScanActivity::class.java)
+            startActivity(intent)
+            finish()
```

Comment:
can we remove this else and just add finish outside?

Reply:
done